### ISSUE TITLE
JSON API and TSV downloads

### ISSUE THREAD

**Issue by sigven** [griffithlab/civic-v2#421]
Hi,

First of all, thanks for an immensely useful resource.

However, as I noticed the upgrade to v2, I have a few questions:

1) I have used the previous(?) JSON API extensively over the past years, and it was a really convenient and easy way to pull down the latest evidence items etc. It seems you have now moved to a new API endpoint, GraphQL API. Does this mean that the JSON api is no longer available? I am confident that the new API is powerful, but I am currently not able to understand how to use this for querying data (variants + evidence items)? Is there a tutorial or something available for this, as a supplement to the technical documentation avilable at https://griffithlab.github.io/civic-v2/?

2) As an alternative to the API, I looked at the downloadable TSV files. From what I can see from browsing, there are now 8,759 evidence items in the database, but I struggle to understand which TSV files can be used to pull down these data. In the the “ClinicalEvidenceSummaries.tsv” (nightly) I found 410 records(?), and no record contained any evidence identifier (i.e. EIDXX). Could you comment on this? The assertions TSV file I was unable to download. Also, I was unable to download any of the TSV files released on the first every month.

kind regards,
Sigve

**Comment by mattchatporter** [comment]
Having the same issue with TSV downloads - more than half of them don't work

**Comment by susannasiebert** [comment]
Hi Sigve and Matt,

Thank you for your interest in CIViC. We'd be happy to help you in switching over your existing API to the new V2 GraphQL API.

The old API is still available under the old url, which now forwards all requests to v1.civicdb.org/api so your existing queries should continue to work as before. Let me know if this doesn't work anymore. One thing to note is that v1 is now in read-only mode so you will still be able to query the data but none of the new data since the switchover will be available.

The new API documentation can be found [here](https://griffithlab.github.io/civic-v2/) and a UI for playing around with the new API is [here](https://civicdb.org/api/graphiql). We recommend the official GraphQL documentation at https://graphql.org/ for familiarizing yourself with GraphQL in general. 

If you can give us an example of which kinds of specific endpoints you used to use and your preferred programming language, we'd be happy to come up with some examples on how to replicate your queries in the new API. We plan on adding new tutorials in the future but haven't gotten around to it at this point.

I will look into the issue with the TSVs.

We appreciate your patience during this switchover period and while we resolve out any bugs we inadvertently introduced.

**Comment by sigven** [comment]
Dear Adam and Susanna,

Thanks a lot for your kind and helpful response. I truly understand that a switchover period will demand some time, I am sorry for raising the issue so quickly :-). 

FYI: Using the old API, i basically just pulled a JSON object directly in using R, e.g. for a given `civic_variant_id`:

`variant_data <- jsonlite::fromJSON(paste0("https://civicdb.org/api/variants/",civic_variant_id)`

Basically, everything I was looking for (variant aliases, chromosomal coordinates, evidence items etc) was found in this object. I am happy to see that it is still working:-) On that note, I will start looking at the GraphQL documentation to understand better how it works. 

kind regards,
Sigve

**Comment by acoffman** [comment]
In addition to just sending a POST request to the API, It looks like there are a number of packages  available in R that allow you to work with GraphQL APIs more ergonomically.

For instance, you could do something like this: https://ropensci.org/blog/2020/12/08/accessing-graphql-in-r/

But with the query swapped out to be `https://civicdb.org/api/graphql` and the query something  like this:

```
{
  variant(id:5){
    name
    id
    description
    variantAliases 
    threePrimeCoordinates  {
      chromosome
      start
      stop
      referenceBases
      variantBases
      representativeTranscript  
    }
    fivePrimeCoordinates {
      chromosome
      start
      stop
      referenceBases
      variantBases
      representativeTranscript  
    }
    evidenceItems {
      totalCount
      pageInfo {
        hasNextPage
        endCursor
      }
      nodes  {
        id
        status
        evidenceDirection
        evidenceType
        evidenceLevel
        evidenceRating
        clinicalSignificance
        description
      	disease {
          id
          doid
          name
        }
        drugs {
          id
          ncitId
          name
        }
        phenotypes {
          id
          hpoId
          name
        }
        drugInteractionType
      }
    }
  }
}
```

**Comment by acoffman** [comment]
If you guys run into any more problems with the downloads or need more guidance on the API, feel free to reach out!
